apps/api/src/modules/providers/sms-plivo/sms-plivo.service.ts

```diff
@@ -27,16 +45,34 @@ export class SmsPlivoService {
   }
 
   async sendMessage(body: SmsPlivoData, providerId: number): Promise<SmsPlivoResponseData> {
-    await this.assignTransport(providerId);
-    const smsPlivoConfig = await this.providersService.getConfigById(providerId);
-    const fromNumber = smsPlivoConfig.PLIVO_SMS_NUMBER as string;
+    try {
+      await this.assignTransport(providerId);
+      const smsPlivoConfig = await this.providersService.getConfigById(providerId);
+      const fromNumber = smsPlivoConfig.PLIVO_SMS_NUMBER as string;
+
+      const response = await this.plivoClient.messages.create({
+        src: fromNumber,
+        dst: body.to,
+        text: body.message,
+      });
 
-    const response = await this.plivoClient.messages.create({
-      src: fromNumber,
-      dst: body.to,
-      text: body.message,
-    });
+      return response;
+    } catch (error) {
+      throw new Error(`Failed to send message: ${error.message}`);
+    }
+  }
+  async getDeliveryStatus(
+    messageUuid: string,
+    providerId: number,
+  ): Promise<PlivoMessageStatusResponse> {
+    try {
+      await this.assignTransport(providerId);
 
-    return response;
+      const response = await this.plivoClient.messages.get(messageUuid);
+      //throw new Error(`Custom error for response: ${JSON.stringify(response)}`);
```

Comment:
Is this needed? Can remove it if not

Reply:
Addressed in https://github.com/OsmosysSoftware/osmo-x/pull/261/commits/5ce4e67bd9877772054c24dac5a90f5ae9c00aad